services/circleci/circleci.service.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const circleSchema = Joi.array()
+  .items(Joi.object({ status: Joi.string() }))
+  .min(1)
+  .required()
+
+module.exports = class CircleCi extends BaseJsonService {
+  async fetch({ token, type, userRepo, branch }) {
+    let url = `https://circleci.com/api/v1.1/project/${type}/${userRepo}`
+    if (branch != null) {
+      url += `/tree/${branch}`
+    }
+    const query = { filter: 'completed', limit: 1 }
+    if (token) {
+      query['circle-token'] = token
+    }
+    return this._requestJson({
+      url,
+      schema: circleSchema,
+      options: { qs: query },
+      errorMessages: { 404: 'project not found' },
+    })
+  }
+
+  static render({ status, color }) {
+    return { message: status, color: color }
```

Comment:
We should turn on [the eslint rule for this](https://eslint.org/docs/rules/object-shorthand); `color: color` can just be `color`.